app/models/mention.rb

```diff
@@ -5,16 +5,22 @@
 class Mention < ActiveRecord::Base
   REGEX = /@\{([^;]+); ([^\}]+)\}/
 
-  belongs_to :post
+  belongs_to :mentions_container, polymorphic: :true
+  belongs_to :post, ->(mention) { (mention.mentions_container_type == "Post") ? where("1 = 1") : none },
```

Comment:
ah, fancy stuff.

But when you do
```ruby
Mention.where(post: Post.find(post_id))
```
it needs two querys and I don't know why this should be better than
```ruby
Mention.where(mentions_container_id: post_id, mentions_contrainer_type: "Post")
```